builtInFunctions/esdtNFTTransfer.go

```diff
@@ -320,6 +320,39 @@ func (e *esdtNFTTransfer) processNFTTransferOnSenderShard(
 	return vmOutput, nil
 }
 
+func migrateNFTMetadataToAccount(
```

Comment:
I think this could be a public function inside esdtDataStorage.go

Reply:
And this only deletes the metadata from system account and changes global settings. It does not actually migrate.

Reply:
Moved there and renamed.  This func removes the metadata from the system account and sets the tokenType as `NonFungibleV2`. Because of the token type, `SaveESDTNFTToken` will save the token metadata to the account, thus the metadata will automatically be migrated.